### PR TITLE
decimal-function

### PR DESCRIPTION
**Pull request by M-OC**

### CODE REVIEW

File: scripts/calculator.js [hunk 1]
```diff
@@ -21,12 +21,31 @@ const addNumber = (num) => {
...
+
+  // if the 'operator' has been set we know we're dealing with the second number
+  if (state['operator'] === '') {
+    currentStep = 'firstNumber'
+  } else {
+    currentStep = 'secondNumber'
+  } 
+
+  if (state[currentStep].indexOf('.') == -1) {
+    (state[currentStep]) += ('.')
```
Comment (M-OC) [inline scripts/calculator.js:41]: The "(" ")" brackets here are unnecessary.

File: scripts/calculator.js [hunk 2]
```diff
@@ -21,12 +21,31 @@ const addNumber = (num) => {
...
+  }  else { 
+    buttonDecimal.disabled = true
```
Comment (M-OC) [inline scripts/calculator.js:43]: `buttonDecimal` is undefined here.

Comment (M-OC) [reply scripts/calculator.js:43]: Assume for the sake of argument that `buttonDecimal` isn't undefined and this disables the "=" button.

What would happen if we were to try to add a decimal to _both_ the first and second numbers?

If a user tries to add a decimal more than once they're effectively locked out; if they try to add it to the second number it'll still be disabled.

File: scripts/calculator.js [hunk 3]
```diff
@@ -61,10 +80,11 @@ const clickListener9 = () => {
 const clickListener0 = () => {
   addNumber('0')
 }
-
-/* Update this so that it also empties state. */
 const clickListenerClear = () => {
   updateScreen()
+  for (let firstNumber in state) {
```
Comment (M-OC) [inline scripts/calculator.js:85]: We should always try to name things what they are.

In this case, the `let` you're declaring here isn't actually "firstNumber". It's a key that'll be a different string each time the loop runs. (once it'll be 'firstNumber', then it'll be 'operator', then 'secondNumber')

If I were writing this, i'd do something along the lines of:
`for (let key in state) {`

File: scripts/calculator.js (hunk 1 above)
Comment (zmaf1989) [reply scripts/calculator.js:41]: Im not sure what this is actually doing, i thought currentStep could represent two different numbers. why would this lock you out of a decimal point on the second number?

Comment (zmaf1989) [reply scripts/calculator.js:41]: I also am not sure of which "('')" to remove, indexOf changes without them i think.

Comment (M-OC) [reply scripts/calculator.js:41]: 1. User adds a '.' to the first number.
2. User tries to add a second '.' to the first number -> button is disabled.
3. User moves to the second number. They try to add a '.' to the second number -> the button is still disabled.

Comment (M-OC) [reply scripts/calculator.js:41]: As far as "(" ")" is concerned, I'm referring to line 41:
`(state[currentStep]) += ('.')`

This does the same thing:
`state[currentStep] += '.'`

Comment (zmaf1989) [reply scripts/calculator.js:41]: is it because it there is nothing re-enabling it?

Comment (M-OC) [reply scripts/calculator.js:41]: Exactly.

**Pull request closed without merging**